echopype/clean/api.py

```diff
@@ -74,3 +85,206 @@ def remove_noise(ds_Sv, ping_num, range_sample_num, noise_max=None, SNR_threshol
+    """
+    Create a mask based on the identified signal attenuations of Sv values at 38KHz.
+    This method is based on:
+    Ryan et al. (2015) ‘Reducing bias due to noise and attenuation in
+        open-ocean echo integration data’, ICES Journal of Marine Science,
+        72: 2482–2493.
+
+    Parameters
+    ----------
+    source_Sv: xr.Dataset or str or pathlib.Path
+        If a Dataset this value contains the Sv data to create a mask for,
+        else it specifies the path to a zarr or netcdf file containing
+        a Dataset. This input must correspond to a Dataset that has the
+        coordinate ``channel`` and variables ``frequency_nominal`` and ``Sv``.
+    mask_type: str with either "ryan" or "fielding" based on
+        the preferred method for signal attenuation mask generation
+    Returns
+    -------
+    xr.DataArray
+        A DataArray containing the mask for the Sv data. Regions satisfying the thresholding
+        criteria are filled with ``True``, else the regions are filled with ``False``.
+
+    Raises
+    ------
+    ValueError
+        If neither ``ryan`` or ``fielding`` are given
+
+    """
+    assert mask_type in ["ryan", "fielding"], "mask_type must be either 'ryan' or 'fielding'"
+    selected_channel_Sv = source_Sv.sel(channel=desired_channel)
+    Sv = selected_channel_Sv["Sv"].values
+    r = source_Sv["echo_range"].values[0, 0]
+    if mask_type == "ryan":
+        # Define a list of the keyword arguments your function can handle
+        valid_args = {"m", "n", "thr", "excludeabove", "operation"}
+        # Use dictionary comprehension to filter out any kwargs not in your list
+        filtered_kwargs = {k: v for k, v in kwargs.items() if k in valid_args}
+        mask = transient_noise._ryan(Sv, r, m=5, **filtered_kwargs)
+    elif mask_type == "fielding":
+        # Define a list of the keyword arguments your function can handle
+        valid_args = {"r0", "r1", "roff", "n", "thr"}
+        # Use dictionary comprehension to filter out any kwargs not in your list
+        filtered_kwargs = {k: v for k, v in kwargs.items() if k in valid_args}
+        mask = transient_noise._fielding(Sv, r, **filtered_kwargs)
+    else:
+        raise ValueError("The provided mask_type must be ryan or fielding!")
+
+    mask = np.logical_not(mask)
+    return_mask = xr.DataArray(
+        mask,
+        dims=("ping_time", "range_sample"),
+        coords={"ping_time": source_Sv.ping_time, "range_sample": source_Sv.range_sample},
+    )
+    return return_mask
+
+
+def get_impulse_noise_mask(
+    source_Sv: xr.Dataset,
+    desired_channel: str,
+    thr: Union[Tuple[float, float], int, float],
+    m: Optional[Union[int, float]] = None,
+    n: Optional[Union[int, Tuple[int, int]]] = None,
+    erode: Optional[List[Tuple[int, int]]] = None,
+    dilate: Optional[List[Tuple[int, int]]] = None,
+    median: Optional[List[Tuple[int, int]]] = None,
+    method: str = "ryan",
+) -> xr.DataArray:
+    """
```

Comment:
A couple suggestions:
- change the input argument to allow users to pass in a dict containing the parameters for the method specified in `method`
- add `frequency_desired` as an input argument to slice the Sv dataset using `frequency_nominal`

Something like below:
```python
def get_impulse_noise_mask(
    source_Sv: xr.Dataset,
    desired_channel: Union[List[str], xr.DataArray],
    method: str = "ryan",
    parameters: Dict,
) -> xr.DataArray
    """
```

Reply:
Also allow variations of type of `source_Sv` as in `get_attentuation_mask`

Reply:
> A couple suggestions:
> * add `frequency_desired` as an input argument to slice the Sv dataset using `frequency_nominal`

Will do - could you explain to me what the logic behind this is? Would this refer to something simialr to https://echopype.readthedocs.io/en/v0.6.1/api/echopype.preprocess.swap_dims_channel_frequency.html ?